### PR TITLE
Disable SNI-DNAT (AUTO_PASSTHROUGH) clusters by default

### LINKED ISSUE #27749: Remove SNI-DNAT clusters by default
These add a substantial performance overhead and many users do not actually use them. We can set it by default in the new eastwest gateway.

cc @rshriram @nmittler would be great to do this for 1.8

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/27749

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by stevenctl — APPROVED** [review]

File: releasenotes/notes/sni-dnat-default.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+apiVersion: release-notes/v2
+kind: feature
+area: networking
+issue:
+- 27749
+releaseNotes:
+- |
+  **Enhanced** the default installation of gateways to not configure clusters for `AUTO_PASSTHROUGH`, reducing memory costs.
```
Comment (rshriram) [inline releasenotes/notes/sni-dnat-default.yaml:8]: may be state it as updated rather than enhanced ?

Comment (howardjohn) [reply releasenotes/notes/sni-dnat-default.yaml:8]: good idea, done

**Review by bianpengyuan — APPROVED** [review]

**Review by morvencao — APPROVED** [review]

**Review by rshriram — APPROVED** [review]

**Review by frankbu — APPROVED** [review]

**Pull request merged**